File: 2023/02 - February/06/03/ChoiceQuestion.java
// ChoiceQuestion.java

/**
 * A question with multiple choices.
 */
public class ChoiceQuestion extends Question {
	/**
	 * Constructs a choice question with a given text and no choices.
	 * @param questionText the text of this question
	 */
	public ChoiceQuestion(String questionText) {
		addLine(questionText);
	}

	/**
	 * Adds an answer choice to this question.
	 * @param choice the choice to add
	 * @param correct true if this is the correct choice, false otherwise
	 */
	public void addChoice(String choice, boolean correct) {
		addLine(choice);
		if (correct) {
			setAnswer(choice);
		}
	}
}

File: 2023/05 - May/08/03/choice_question.py
from question import Question

# Choice Question Class
class ChoiceQuestion(Question):
	# Constructor
	def __init__(self):
		# A List To Hold All The Choices
		self.choices = []
	
	# Adds A New Choice To Question
	def add_choice(self, choice, correct):
		# Add Choice To List Of Choices
		self.choices.append(choice)
		# If Correct Flag Is Set, This Is The Correct Answer
		if correct:
			# Set The Answer To The Choice Index
			self.answer = str(len(self.choices))

	# Returns A String Of Question Along WithAll Choices
	def __str__(self):
		# Append Question
		question = self.question
		# Append All Choices
		for i in range(len(self.choices)):
			question += f'\n{i + 1}. {self.choices[i]}'
		# Return The Result
		return question
	